### PR TITLE
[WIP] Add extension Nbody part

### PR DESCRIPTION
**Pull request by npeschken**
* Move particles to cgs 
* Parallelize particles
* IO of particles with restart files
* rebuild particle data structure into lists

### CODE REVIEW

**Comment by gawrysz** [comment]
Can one of the admins verify this patch?

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by woolt** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
test this please

**Comment by gawrysz** [comment]
test this please

**Comment by gawrysz** [comment]
test this please

**Comment by gawrysz** [comment]
test this please

**Comment by gawrysz** [comment]
test this please

**Comment by gawrysz** [comment]
test this please

**Comment by gawrysz** [comment]
Jenkins test this please

File: src/base/piernik.F90 [hunk 1]
```diff
@@ -169,19 +169,20 @@ program piernik
          ! It can be used for diagnostic purposes. In production runs it may cost too much.
          if (try_rebalance) then
             !> \todo try to rewrite this ugly chain of flags passed through global variables into something more fool-proof
-            call leaves%balance_and_update(" (re-balance) ")
             call all_bnd ! For some strange reasons this call prevents MPI-deadlock
             try_rebalance = .false.
          endif
+         call ppp_main%stop('try_reb')
```
Comment (gawrysz) [inline src/base/piernik.F90:173]: stop without start

File: src/base/piernik.F90 [hunk 2]
```diff
@@ -169,19 +169,20 @@ program piernik
          ! It can be used for diagnostic purposes. In production runs it may cost too much.
          if (try_rebalance) then
             !> \todo try to rewrite this ugly chain of flags passed through global variables into something more fool-proof
-            call leaves%balance_and_update(" (re-balance) ")
```
Comment (gawrysz) [inline src/base/piernik.F90:169]: why it got deleted?

**Comment by gawrysz** [comment]
Jenkins test this please

**Comment by gawrysz** [comment]
Test this please

**Comment by gawrysz** [comment]
Jenkins test this please

**Review by woolt — APPROVED** [review]

**Pull request merged**